src/messaging/ExchangeContext.cpp

```diff
@@ -309,6 +308,10 @@ bool ExchangeContext::MatchExchange(SecureSessionHandle session, const PacketHea
         // AND The message was received from the peer node associated with the exchange
         && (mSecureSession == session)
 
+        // This check should be already implied by the equality of session check,
+        // It should be removed after we have implemented the temporary node id for PASE and CASE sessions
+        && (IsEncryptionRequired() == packetHeader.GetFlags().Has(Header::FlagValues::kEncryptedMessage))
```

Comment:
Moving this check here will bypass the check when UMH is used for receiving the message.

Reply:
UMH part is also checked https://github.com/project-chip/connectedhomeip/pull/8726/files#diff-89ece36db3e262edfa7393ade54f24c5ade45c69931a282ec4af5ee30143e1a3R293-R301